http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -415,6 +427,81 @@ object ClusterTemplateValues {
         .getOrElse(""),
       clusterResourceConstraints.map(_.memoryLimit.toString).getOrElse("")
     )
+
+  def fromCreateCluster(createCluster: CreateCluster,
+            initBucketName: Option[GcsBucketName],
+            stagingBucketName: Option[GcsBucketName],
+            serviceAccountKey: Option[ServiceAccountKey],
+            dataprocConfig: DataprocConfig,
+            welderConfig: WelderConfig,
+            proxyConfig: ProxyConfig,
+            clusterFilesConfig: ClusterFilesConfig,
+            clusterResourcesConfig: ClusterResourcesConfig,
+            clusterResourceConstraints: Option[ClusterResourceConstraints]): ClusterTemplateValues =
+    ClusterTemplateValues(
+      createCluster.clusterProjectAndName.googleProject.value,
+      createCluster.clusterProjectAndName.clusterName.value,
+      stagingBucketName.map(_.value).getOrElse(""),
+      createCluster.clusterImages.find(_.imageType == Jupyter).map(_.imageUrl).getOrElse(""),
+      createCluster.clusterImages.find(_.imageType == RStudio).map(_.imageUrl).getOrElse(""),
+      proxyConfig.jupyterProxyDockerImage,
+      createCluster.clusterImages.find(_.imageType == Welder).map(_.imageUrl).getOrElse(""),
+      initBucketName
+        .map(n => GcsPath(n, GcsObjectName(clusterFilesConfig.jupyterServerCrt.getName)).toUri)
```

Comment:
can this be refactored into a function that takes the relevant name from the config?

i.e. 
```
 initBucketName
        .map(n => GcsPath(n, GcsObjectName(clusterResourcesConfig.jupyterDockerCompose.value)).toUri)
        .getOrElse("")
```

turns to 

```
initBucketNameToUri(initBucketName, clusterResourcesConfig.jupyterDockerCompose.value)
```

It would look cleaner as one line instead of 3 in my opinion.

Reply:
this is direct copy from existing code, which Rob will have some refactoring so I'd like to keep it as is so that later when we merge his PR, it's more obvious that this code can be refactored the exact same way

Reply:
Seems reasonable to me, I am refactoring `ClusterTemplateValues` to its own file